flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputChannel.java

```diff
@@ -276,21 +277,36 @@ protected int getCurrentBackoff() {
      * @return <code>true</code>, iff the operation was successful. Otherwise, <code>false</code>.
      */
     protected boolean increaseBackoff() {
+        return increaseBackoff(0);
+    }
+
+    /**
+     * The remote task manager creates partition request listener and returns {@link
+     * PartitionNotFoundException} until the listener is timeout, so the backoff should add the
+     * timeout milliseconds if it exists.
+     *
+     * @param timeoutMS The timeout milliseconds that the partition request listener timeout
+     * @return <code>true</code>, iff the operation was successful. Otherwise, <code>false</code>.
+     */
+    protected boolean increaseBackoff(int timeoutMS) {
```

Comment:
I see that you kept the original logic of the backoff and just taken into the account this timeout. Are you sure this is the right logic? What if the request fails immediately, without waiting for this `partition-request-notify-timeout-ms`, because upstream TM hasn't yet started? Wouldn't it mean that the backoffs would not work at all, and the requests would fail within a couple of ms after a couple of retries?